chainimport/headers_import.go

```diff
@@ -0,0 +1,1403 @@
+// don't have access to the actual compact filters that were used to generate
+// these filter headers. Filter headers are calculated using the formula:
+//
+// FilterHeader_N = double-SHA256(FilterHeader_N-1 || double-SHA256(Filter_N))
+//
+// where FilterHeader_N-1 is the previous filter header in little-endian byte
+// order, Filter_N is the compact filter for the block, and || represents
+// concatenation.
+func (v *FilterHeadersImportSourceValidator) Validate(
+	iterator HeaderIterator[*ChainImportFilterHeader],
+	targetChainParams chaincfg.Params) error {
+
+	return nil
+}
+
+// ValidatePair checks if two consecutive filter headers maintain the correct
+// cryptographic relationship.
+//
+// In a full validation, we would verify that each filter header is correctly
+// derived by hashing the previous filter header with the current block's
+// compact filter:
+//
+// However, during import, we don't have access to the compact filters, so we
+// can't validate if two consecutive filter headers maintain the correct
+// cryptographic relationship.
+func (v *FilterHeadersImportSourceValidator) ValidatePair(prev,
+	current *ChainImportFilterHeader, targetChainParams chaincfg.Params) error {
+
+	return nil
+}
+
+// HeadersImport uses a generic HeaderSource to import headers.
+type HeadersImport struct {
+	BlockHeadersImportSource  HeaderImportSource[*ChainImportBlockHeader]
+	FilterHeadersImportSource HeaderImportSource[*ChainImportFilterHeader]
+	BlockHeadersValidator     HeadersValidator[*ChainImportBlockHeader]
+	FilterHeadersValidator    HeadersValidator[*ChainImportFilterHeader]
+}
+
+// Import is a multi-pass algorithm that loads, validates, and processes
+// headers from the configured import sources into the target header stores. The
+// Import process is currently performed only if the target stores are
+// completely empty except for gensis block/filter header otherwise it is
+// entirely skipped.
+func (s *HeadersImport) Import(ctx context.Context,
+	options ImportOptions) (*ImportResult, error) {
+
+	// Check first if the target header stores are fresh.
+	isFresh, err := s.isTargetFresh(
+		options.TargetBlockHeaderStore,
+		options.TargetFilterHeaderStore,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("headers import failed: %w", err)
+	}
+	if !isFresh {
+		log.Info("Skipping headers import: target header stores are " +
+			"not empty")
+		return &ImportResult{}, nil
+	}
+
+	// Create result structure at the beginning to capture total duration.
+	result := &ImportResult{
+		StartTime: time.Now(),
+	}
+
+	// Initialize and open header sources before proceeding with validation
+	// and import operations.
+	if err := s.initSources(); err != nil {
+		return nil, fmt.Errorf("headers import failed: %w", err)
+	}
+	defer s.deinitSources()
+
+	// Check if the sources are compatible with each other.
+	if err := s.validateSourcesCompatibility(
+		s.BlockHeadersImportSource, s.FilterHeadersImportSource,
```

Comment:
So in some methods of `HeadersImport` we access these two fields directly. But here we pass them as parameters.
I think we should do the same everywhere and unify the method signatures a bit.
Probably it also makes sense to add the options as a field to the `HeadersImport` struct, so it can be accessed in the methods directly..